### PR TITLE
Improve DAG Text Formatting

### PR DESCRIPTION
**Pull request by plypaul**
### Description

This PR:

* Adds `Dag.text_structure() / DagNode.text_structure()` to make it easier to generate the text representation of a DAG.
* Removes previous methods that were used for text formatting a DAG.
* Updates the logic for how the max. column width was handled when formatting a DAG to reduce the column width of snapshot files.

### CODE REVIEW

File: metricflow/test/snapshots/test_conversion_metrics_to_sql.py/SqlQueryPlan/test_conversion_count_with_no_group_by__plan0.xml [hunk 1]
```diff
@@ -24,102 +30,207 @@
...
+                <!--     expr=SqlAggregateFunctionExpression(node_id=fnc_0, sql_function=SUM),  -->
+                <!--     column_alias='visits',                                                 -->
+                <!--   )                                                                        -->
                 <!-- from_source = SqlSelectStatementNode(node_id=ss_10) -->
                 <!-- where = None -->
                 <!-- distinct = False -->
                 <SqlSelectStatementNode>
-                    <!-- description = Pass Only Elements: ['visits',] -->
-                    <!-- node_id = ss_10 -->
+                    <!-- description = "Pass Only Elements: ['visits',]" -->
```
Comment (courtneyholcomb) [inline metricflow/test/snapshots/test_conversion_metrics_to_sql.py/SqlQueryPlan/test_conversion_count_with_no_group_by__plan0.xml:44]: Should we default to double quotes for these descriptions for better consistency? Not a strong opinion here.

**Review by courtneyholcomb — APPROVED** [review]
LGTM! Thanks!

File: metricflow/test/snapshots/test_conversion_metrics_to_sql.py/SqlQueryPlan/test_conversion_count_with_no_group_by__plan0.xml (hunk 1 above)
Comment (plypaul) [reply metricflow/test/snapshots/test_conversion_metrics_to_sql.py/SqlQueryPlan/test_conversion_count_with_no_group_by__plan0.xml:44]: Do you mean how it uses `"` in some cases but `'` in others? It seems that's the behavior of `repr()` / `pprint.pformat()` if it's a string or a string containing a quote.

File: metricflow/dag/mf_dag.py [hunk 2]
```diff
@@ -111,6 +112,10 @@ def accept_dag_node_visitor(self, visitor: DagNodeVisitor[VisitorOutputT]) -> Vi
         """Visit this node."""
         return visitor.visit_node(self)
 
+    def text_structure(self, formatter: MetricFlowDagTextFormatter = MetricFlowDagTextFormatter()) -> str:
```
Comment (tlento) [inline metricflow/dag/mf_dag.py:115]: This change warms my cold grinchy heart....

**Pull request merged**